pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,350 @@
+		Value(machinePoolTaintValue)
+}
+
+func createNewMachinePoolLabelsDedicated() (*v1.Label, error) {
+	label := v1.NewLabel().
+		Key(machinePoolLabelKey).
+		Value(machinePoolLabelValue)
+	l, err := label.Build()
+	if err != nil {
+		return nil, err
+	}
+	return l, nil
+}
+
+func createMachinePoolLabelMap(labels *v1.Label) map[string]string {
+	labelMap := make(map[string]string)
+	labelMap[labels.Key()] = labels.Value()
+	return labelMap
+}
+
+func createMachinePoolRequestForDedicated(machinePoolNodeCount int) (*v1.MachinePool, error) {
+	labels, err := createNewMachinePoolLabelsDedicated()
+	if err != nil {
+		return nil, err
+	}
+	mp := v1.NewMachinePool()
+	mp.ID(machinePoolId).
+		Replicas(machinePoolNodeCount).
+		InstanceType(machinePoolInstanceType).
+		Labels(createMachinePoolLabelMap(labels)).
+		Taints(createNewMachinePoolTaintsDedicated())
+	machinePool, err := mp.Build()
+	if err != nil {
+		return nil, err
+	}
+	return machinePool, nil
+}
+
+// TODO this function should be moved to an ocm client / provider area
+func createMachinePool(opts *options, mprequest *v1.MachinePool) error {
+	// create a new machine pool via ocm
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, err := conn.API().OCMClustermgmt()
+	if err != nil {
+		return err
+	}
+	response, err := client.Clusters().Cluster(opts.selectedCluster.ID()).MachinePools().Add().Body(mprequest).Send()
+	if err != nil {
+		return err
+	}
+	opts.selectedClusterMachinePool = *response.Body()
+	return nil
+}
+
+func createMachinePoolInteractivePrompt(opts *options) error {
+	validator := &dedicatedcmdutil.Validator{
+		Localizer:  opts.f.Localizer,
+		Connection: opts.f.Connection,
+	}
+	// TODO add page size and Localizer
+	promptNodeCount := &survey.Input{
+		Message: "Enter the desired machine pool node count",
+		Help:    "The machine pool node count must be greater than or equal to and a multiple of 3",
+	}
+	var nodeCount int
+	err := survey.AskOne(promptNodeCount, &nodeCount, survey.WithValidator(validator.ValidatorForMachinePoolNodes))
+	if err != nil {
+		return err
+	}
+	opts.requestedMachinePoolNodeCount = nodeCount
+	dedicatedMachinePool, err := createMachinePoolRequestForDedicated(nodeCount)
+	if err != nil {
+		return err
+	}
+	err = createMachinePool(opts, dedicatedMachinePool)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// machine pool replica count must be greater than or equal and a multiple of 3
+func validateMachinePoolNodes(opts *options) error {
+	if len(opts.clusterMachinePoolList.Slice()) > 0 {
+		for _, machinePool := range opts.clusterMachinePoolList.Slice() {
+			if validateMachinePoolCount(machinePool.Replicas()) &&
+				checkForValidMachinePoolLabels(*machinePool) &&
+				checkForValidMachinePoolTaints(*machinePool) {
+				opts.selectedClusterMachinePool = *machinePool
+				return nil
+			} else {
+				err := createMachinePoolInteractivePrompt(opts)
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
This should be done only when we cannot find a machine pool that matches the criteria i.e outside of the loop